ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -189,52 +187,63 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-
-            if (scroll) {
-                this.scrollToTop();
-            }
-
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-
-                this.isPageBusy.pipe(
-                    tap(_ => this.isLoaded$.next(false)),
-                    debounceTime(this.timeToDebounce),
-                    filter(busy => !busy),
-                    take(1)
-                ).subscribe(_ => {
-                    this.isLoaded$.next(true);
-                    this.currentSectionIndex = nextIndex;
-                    this.visitedSectionIndexes[nextIndex] = true;
-                    this.saveLastVisitedSectionIndex(nextIndex);
-                });
-            }
+            this.submitAnnouncementService.announceSubmit();
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(_ => this.isLoaded$.next(false)),
+                        tap(() => {
+                            // run validations and compute flags for blocks to enable scrolling to errors
+                            this.currentSection.validate();
+                            this.model && this.model.shouldScrollToFirstInvalidQuestion();
+                            // triggers scrolling
+                            this.validationRequested = true;
+                        }),
+                        // delay needed for validationRequested to be processed
+                        delay(10),
+                        tap(() => {
+                            this.sendSectionAnalytics();
+                            tap(_ => this.isLoaded$.next(true)),
```

Comment:
Incorrectly placed tap (into another tap)

Reply:
Yes. This was not good!
a9c206014